src/ir/attributes/double_attribute.h

```diff
@@ -0,0 +1,60 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#ifndef SRC_IR_ATTRIBUTES_DOUBLE_ATTRIBUTE_H_
+#define SRC_IR_ATTRIBUTES_DOUBLE_ATTRIBUTE_H_
+
+#include "absl/strings/str_format.h"
+#include "src/common/utils/intrusive_ptr.h"
+#include "src/ir/attributes/attribute.h"
+
+namespace raksha::ir {
+
+class DoubleAttribute;
+bool operator==(const DoubleAttribute& lhs, const DoubleAttribute& rhs);
+
+class DoubleAttribute : public AttributeBase {
```

Comment:
Not for this PR, but a general observation. It seems like int and double attributes could be templatized.